### DISCUSSION TITLE
How can I run euphony-hub-collector with android emulator?

### DISCUSSION THREAD

**Discussion by yeonns** [euphony-io/euphony#134] [Q&A] [answered]
Hi, I don't have any real android devices, so I have to run ```euphony-hub-collector``` with emulator.
But, when I launch the app and type "hi", the progress bar keeps spinning like below! :cry: 

How can I get the result with emulator?
![스크린샷, 2022-07-10 13-53-29](https://user-images.githubusercontent.com/10149398/178131821-4aae44e6-2c4b-47de-a376-c05fe4e3fc9d.png)

## My Emulator Setting is...
- Category: Phone
- API version: 30
- RAM: 1536MB
- VM heap: 384MB
- Internal Storage: 800MB

**Comment by win-luck** [comment, marked as answer]
hello. In fact, this is my problem. from yesterday to today's afternoon , I was trying to construct Euphony's sample project in Android studio. but too many error make me sad. after all, I had to run Emulator(AVD device) because my phone is iPhone, but Emulator setting forced a lot of time to fix this problem. Today, I was listening the Git education of OSSCA and fixing my error with yeonns at the same time. It was confused. Luckliy, I solved my problem. I really appreciate her. 

The original File wasn't work in my Android studio, So I used modified version of yeonns.

first, you should select path of the File, Open the git bash and write
$ git remote add yeonns https://github.com/yeonns/euphony-hub-collector 

$ git pull yeonns layout

$ git log -5 --pretty=oneline

![제목 없음](https://user-images.githubusercontent.com/53044069/178141278-ce1be798-8d8a-4c92-a2e0-dcc474c7581c.png)
if you got "implement audio recorder" at first message, you're right.


you must check these options before build the project. (**"Euphony_collector" by kotlin**)
![image](https://user-images.githubusercontent.com/53044069/178140204-4ccff7a0-36d3-4cc9-a3fd-2226b6ad829f.png)
first, select Virtual Device's API version "30". I selected API "31 to 33", but the problem wasn't solved. i don't know why.
![image](https://user-images.githubusercontent.com/53044069/178140351-7b0133a3-0b74-487f-b12d-194416235dfa.png)
and click "Extended Controls"
![image](https://user-images.githubusercontent.com/53044069/178140395-b6679f78-86f9-40f0-9403-6a79ef339972.png)
Go [microphone] and set options like this image.

Set PC and Emulator's volume max. You will hear little sound. It's soundwave.  And,,,
![image](https://user-images.githubusercontent.com/53044069/178140534-2434f3d3-e7d6-4e10-8d80-75038e0a1653.png)
 You will get the message "success"

You may get Errors like these:
"The emulator process for AVD has terminated"
-> https://codingjerk-diary.tistory.com/entry/Android%EC%98%A4%EB%A5%98%ED%95%B4%EA%B2%B0-The-emulator-process-for-AVD-has-terminated
"AVD is already running"
-> https://m.blog.naver.com/PostView.naver?isHttpsRedirect=true&blogId=pyj721aa&logNo=221282430278
"AVD's loading isn't finishing"
-> File - Settings - Tool - Emulator - Uncheck "launch in a tool window"

I refered this page to solve the problem. I hope it will be helpful for us.

**Of course, you don't have to refer this if you have an android device.**

**Reply by designe** [reply]
Thanks for super detail answer! :)
It is the issue for abnormal operation on the emulator from API version 31 to 33.

We need to make issue for this. and find a solution!

**Comment by yeonns** [comment]
@chaso1495 Thank you for detailed screenshot and explanation!

In fact, I thought that ```euphony-hub-collector``` has problem at first, so I suggested you to pull my repository, but cloning original repository would work :smile: 
So, instead of below commands:
```
$ git remote add yeonns https://github.com/yeonns/euphony-hub-collector
$ git pull yeonns layout
$ git log -5 --pretty=oneline
```
run this:
```
$ git clone https://github.com/euphony-io/euphony-hub-collector
```

Your answer will be very helpful for those who want to run ```euphony-hub-collector``` with emulator. :+1: 
Thank you! :heart:

**Comment by sunyeongan** [comment]
It helped a lot. It would be better if the emulator setting part was added to docs.👍

**Reply by designe** [reply]
great idea ;)
